### PR TITLE
Add OpenDirector.addheaders and PEP 526 refactoring (fixes #2163)

### LINKED ISSUE #2163: urllib.request.OpenerDirector missing addheaders attribute
`OpenDirector.addheaders` is referred to in https://docs.python.org/3/library/urllib.request.html#examples. However, the stubs do not include it:

https://github.com/python/typeshed/blob/2ba90a65c0cf4d196a971e1c0b0362bb735e8e6d/stdlib/3/urllib/request.pyi#L70-L74

Note that older stubs for Py2.x do in at least one place:

https://github.com/python/typeshed/blob/100a071729758207775fda2ee5c0982bc7deae2c/stdlib/2/urllib.pyi#L13-L19

Happy to submit a 1-line PR if that's not too small of a change:

```diff
diff --git a/stdlib/3/urllib/request.pyi b/stdlib/3/urllib/request.pyi
index 23e6e91..486a758 100644
--- a/stdlib/3/urllib/request.pyi
+++ b/stdlib/3/urllib/request.pyi
@@ -68,6 +68,7 @@ class Request:
     def header_items(self) -> List[Tuple[str, str]]: ...

 class OpenerDirector:
+    addheaders = ...  # type: List[Tuple[str, str]]
     def add_handler(self, handler: BaseHandler) -> None: ...
     def open(self, url: Union[str, Request], data: Optional[bytes] = ...,
              timeout: float = ...) -> _UrlopenRet: ...
```

I did not notice any other similar missing attributes for `OpenerDirector` from the docs and the source doesn't have any others that should unambiguously be public.

### PR DESCRIPTION
**Pull request by sushain97**

### CODE REVIEW

File: stdlib/3/urllib/request.pyi [hunk 1]
```diff
@@ -68,14 +68,15 @@ class Request:
     def header_items(self) -> List[Tuple[str, str]]: ...
 
 class OpenerDirector:
+    addheaders: List[Tuple[str, str]] = ...
```
Comment (JelleZijlstra) [inline stdlib/3/urllib/request.pyi]: Strictly speaking this should not have `= ...` because it does not exist on the class, only on instances.

Comment (sushain97) [reply stdlib/3/urllib/request.pyi]: Ah, I didn't realize that's how the distinction manifested in the stubs. In that case, it seems that a number of `Request`'s attributes should also not have `= ...` (all)?

https://github.com/python/cpython/blob/3.6/Lib/urllib/request.py#L324

Comment (JelleZijlstra) [reply stdlib/3/urllib/request.pyi]: Yes, probably most (I didn't look through all of them).

I'm not sure type checkers currently implement this distinction, but they are supposed to. `x: int` in class scope means "instance variable only";  `x: int = ...` means "instance variable, with a default value on the class", and `x: ClassVar[int]` means "class variable only".

**Comment by JelleZijlstra** [comment]
Thank you!

**Pull request merged**